tests/rptest/redpanda_cloud_tests/high_throughput_test.py

```diff
@@ -897,17 +913,17 @@ def test_decommission_and_add(self):
 
         # Generate a realistic number of segments per partition.
         self.load_many_segments()
-        producer = None
-        try:
-            producer = KgoVerifierProducer(
```

Comment:
Issue found by type checking. This assignment should be outside the try (and the `None` assignment removed), since otherwise if the construction throws we will deference None in the handler. This same bug repeats several times below. Found by type checking.